iprate.go

```diff
@@ -444,15 +456,22 @@ func sendImmediately(guarantee time.Duration) bool {
 
 // Removes an IP from ratelimits
 func (ipr *ipRate) RemoveLimit() {
-	ipr.RLock()
-	defer ipr.RUnlock()
 	if !noop {
+		ipr.RLock()
 		for service, _ := range aclByService {
 			if ipr.limitedOnService[service.ID] {
 				msg := limitMessage{service: service, ipRate: ipr, operation: fastly.BatchOperationDelete}
 				limitCh <- &msg
 			}
 		}
+		ipr.RUnlock()
+	}
+
+	if noop {
```

Comment:
Decided to lock inside because in one case I'm only reading, in the second case I'm writing.

Will address the `else`.